core/src/validator.rs

```diff
@@ -168,7 +170,44 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let kp = keypair.clone();
+        let mut counter = 0;
+        let counter_ref = &mut counter;
+        if counter == 0 {
+            set_handler(move || { 
+                *counter_ref += 1;
+                let bank = bf.read().unwrap().working_bank();
+                let validator_pubkey = kp.pubkey();
+                let mut current_stakes: Vec<_>  = bank
+                .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0)
+                .map(|accounts| {
+                    accounts
+                        .iter()
+                        .filter_map(|(pubkey, (stake, account))| {
```

Comment:
nit: I think we can make this block smaller, like this:
```rust
                        .filter_map(|(pubkey, (stake, account))| {
                            if let Ok(vote_state) = VoteState::deserialize(&account.data) && vote_state.node_pubkey == validator_pubkey { 
                                  return Some((*pubkey, *stake));
                            }
                            None
                        })
```

Reply:
Does this approach not work well?

Reply:
What do you mean here?

Reply:
I'm just suggesting the `filter_map()` implementation be restructured to avoid the double `else { None }` at the end, with the sample code I wrote